main.tf

```diff
@@ -71,13 +72,17 @@ data "aws_iam_policy_document" "secret_access_policy_doc" {
 
 data "vault_generic_secret" "vault_secret" {
   for_each = { for secret_meta in var.vault_secrets : secret_meta.env_name => secret_meta }
-  path     = each.value.path
+  # Force the data source to be looked up during the apply step, not the plan step.
+  # See: https://github.com/hashicorp/terraform-provider-vault/issues/1221
+  path     = trimprefix("${local.always_changing_value}${each.value.path}", local.always_changing_value)
```

Comment:
Very cool! It's a bit hard to understand without context, but should be good for now. Do you anticipate similar issues with Pulumi "applies"?

Reply:
@dipack95 Unfortunately this doesn't work from a test. There seems to be a bug in the vault provider that reuses old credentials from a planfile.

Pulumi would not have this problem, but only because it doesn't support planfiles as a feature. If we don't use planfiles in terraform, this problem goes away